distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso

```diff
@@ -347,7 +402,7 @@ post (uri:(URI | Text)=(Missing_Argument.throw "uri")) (body:Request_Body=..Empt
 download : (URI | Text) -> Writable_File -> HTTP_Method -> Vector (Header | Pair Text Text) -> File ! Request_Error | HTTP_Error
 download (uri:(URI | Text)=(Missing_Argument.throw "uri")) file:Writable_File (method:HTTP_Method=..Get) (headers:(Vector (Header | Pair Text Text))=[]) =
     Context.Output.if_enabled disabled_message="As writing is disabled, cannot download to a file. Press the Write button ▶ to perform the operation." panic=False <|
-        response = HTTP.fetch uri method headers
+        response = HTTP.fetch uri method headers cache_policy=Cache_Policy.Dont_Cache_Responses
```

Comment:
Could we rename this option to `No_Cache`? Then it would be clear that it is not saving cached responses, nor using existing cache.

I think that controlling if response is saved to cache is not very useful. BUT we want to control if the cache is _used_ - I want to be able to specify `No_Cache` and then have a guarantee that a new request is performed, omitting the cache.

Reply:
@jdunkerley with `Dont_Cache_Responses` it doesn't write to or read from the cache. I don't think it's worth complicating the logic for this case, which is when a user fetched or downloaded with caching on and then turned caching off. Turning caching off suggests they always want fresh data. But I can be convinced otherwise.

@radeusgd the reason this option prevents trying to write to the cache is for when the user wants data that is too large for the limits set in the cache. But we could cache the data if it isn't too large, even if the user wants the data checked for freshness.

This does bring up the question of re-validating data (the "no-cache" and "must-revalidate" response headers) -- but is this something we want users to deal with? We could have the following, which follow HTTP naming conventions:

```
type Cache_Policy
    ## Use the cache while the data is still fresh, then request new data. If you can't, keep using the cache.
    Default
    ## you must revalidate every time, and if you can’t, fail.
    No_Cache
    ## you must revalidate when the data becomes stale, and if you can’t, fail.
    Must_Revalidate
    ## never store data in the cache
    No_Store
```

(The `No_Store` one makes more sense for a web cache, for security reasons, but in our case, the cache and the engine are always in the same running instance.)

So, we could have all these options. If it seems too complicated, then I think a simple on/off is best -- either you cache, or you always make the request.

Reply:
I agree with your perspective, I think having too many options will complicate stuff unnecessarily. I think cache on/off setting is much easier to understand than splitting this into numerous variants.

I just wanted to suggest renaming `Dont_Cache_Responses` into something like `No_Cache`. Because IMO `Dont_Cache_Responses` sounds like you won't save the response, but if it's already cached, it is not clear if cached version will be used or not. We agree it should not be used as users need a way to force fresh data. So it would be good to name this setting in a way that will make that clear for the user. IMO `Dont_Cache_Responses` does not make it clear. But `No_Cache` or `No_Caching` or something similar, is a bit clearer.